server/src/main/java/org/apache/iotdb/db/engine/StorageEngine.java

```diff
@@ -950,8 +950,23 @@ public boolean unloadTsfile(File tsfileToBeUnloaded, File targetDir)
    * @param file internal file
    * @return sg name
    */
-  public String getSgByEngineFile(File file) {
-    return file.getParentFile().getParentFile().getParentFile().getName();
+  public String getSgByEngineFile(File file, boolean needCheck) throws IllegalPathException {
```

Comment:
add javadoc for needCheck

like: check if the tsfile is an internal TsFile, if you make sure it is inside, no need to check